### ISSUE TITLE
Accordion - Code as a button instead of tab?

### ISSUE THREAD

**Issue by HerinHentry** [govau/uikit#395]
Thanks for the great work on the Design System components. I am trying to review them from a users perspective and found an ARIA specification related error.

Accordions are created as with role="tab". But, as per the ARIA spec, role="tab", should be contained in the role="tablist"

Page: https://designsystem.gov.au/components/accordion/
Role spec: https://www.w3.org/WAI/PF/aria/roles#tab

Current code:
  <a href="#accordion-default" class="au-accordion__title js-au-accordion" aria-controls="accordion-default" aria-expanded="true" aria-selected="true" **role="tab"** onclick="return AU.accordion.Toggle( this )">
    Accordion title
  </a>

Thanks,
Herin

**Comment by alex-page** [comment]
Thanks @HerinHentry this seems really valid. I am thinking we should remove the aria `role="tab"` as we are not sure if there is going to be a group or just an individual accordion. We can then replace it with something like:

http://heydonworks.com/practical_aria_examples/#progressive-collapsibles

**Comment by HerinHentry** [comment]
Thanks @alex-page. The W3c's example seems to be a good one.
https://www.w3.org/TR/wai-aria-practices-1.1/examples/accordion/accordion.html

And also, they are using buttons and not links. It's worth thinking if we want to have them as buttons / links?

**Comment by HerinHentry** [comment]
Also, the menu and the submenu on the left are coded as Tabs.
![image](https://user-images.githubusercontent.com/15111528/40337594-c9d806c0-5db3-11e8-8e85-b172be464fa1.png)
I think having them as menu indicate that these are navigational elements and not process elements.
https://www.w3.org/TR/wai-aria-practices-1.1/#menu

**Comment by alex-page** [comment]
@HerinHentry Thanks for picking that up, definitely valid to change those in that scenario.

**Comment by shad0wMe0w** [comment]
Should accordions always be assigned a **tab** role, or is it still okay to create them as <button> nested in headings?

**Comment by HerinHentry** [comment]
Creating them as buttons is more meaningful. When accordions are used to indicate progress or forms, it's easy to control their states as active/disabled when they are buttons. If they are coded as links, <a> do not have a disabled property.
No need for role="tab".

**Comment by ronsman** [comment]
I agree with Herin. See [https://www.w3.org/TR/wai-aria-practices/#accordion](https://www.w3.org/TR/wai-aria-practices/#accordion)

**Comment by alex-page** [comment]
The issue this ticket raises is that a link should go to somewhere on a page. The accordion link currently is a reference to the content directly below the title. It works like an in-page navigation. 

This potentially can be confusing for a user. Currently without javascript we can be sure that the relationship still exists, however using the correct aria tags we may be able to do the same thing.

I am going to bring this up with our accessibility experts as I believe switching to a button with the correct aria tags will provide the same solution without the confusion of a link can bring to a user. I am also interested if the current solution makes sense or if it still be confusing for a user. 

If anyone has any thoughts feel free to post them here 👍

**Comment by alex-page** [comment]
Thanks @HerinHentry we have resolved this issue and are going to be updating our documentation shortly.
